src/QuoteHelper.ts

```diff
@@ -189,24 +190,27 @@ export class QuoteHelper {
   /**
    * Decode an oracle context payload into signed quotes.
    * @param oracleContext Oracle context
-   * @param version Quote version (1 or 2)
+   * @param version Quote version ('1' or '2')
    * @return Signed quotes
    */
-  static decodeQuotes(oracleContext: `0x${string}`, version: number = 1): readonly SignedQuote[] {
-    if (version === 1) {
+  static decodeQuotes<T extends '1' | '2' = '1'>(
+    oracleContext: `0x${string}`,
+    version: T = '1' as T,
+  ): T extends '1' ? readonly SignedQuote<QuoteV1>[] : readonly SignedQuote<QuoteV2>[] {
```

Comment:
this could be an overkill or hard to read, let me know if I should remove, or we can declare a type and change to
```diff
-  ): T extends '1' ? readonly SignedQuote<QuoteV1>[] : readonly SignedQuote<QuoteV2>[] {
+  ): decodeQuotesReturnType<T> {

-  )[0] as T extends '1' ? readonly SignedQuote<QuoteV1>[] : readonly SignedQuote<QuoteV2>[];
+  )[0] as decodeQuotesReturnType<T>;
+ type decodeQuotesReturnType<T extends '1' | '2' = '1'> = T extends '1' ? readonly SignedQuote<QuoteV1>[] : readonly SignedQuote<QuoteV2>[];
```